libraries/AP_Networking/AP_Networking.cpp

```diff
@@ -107,7 +107,9 @@ void AP_Networking::init()
 
     if (!backend->init()) {
         GCS_SEND_TEXT(MAV_SEVERITY_INFO, "NET: backend init failed");
+        delete backend;
```

Comment:
i think instead we add a comment explaining why we don't delete
we could have callbacks or interrupt handlers installed, or a thread relying on the backend